app/services/evss_claim_service_async.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class EVSSClaimServiceAsync
+  include SentryLogging
+
+  def initialize(user)
+    @user = user
+    @tracker = EVSSClaimsSyncStatusTracker.new(user_uuid: user.uuid)
+  end
+
+  def all
+    status = @tracker.get_collection_status
+    unless status
+      status = 'REQUESTED'
+      @tracker.set_collection_status(status)
+      EVSS::RetrieveClaimsFromRemoteJob.perform_async(@user.uuid)
+    end
+    claims = status == 'REQUESTED' ? [] : claims_scope.all
+    @tracker.delete_collection_status unless status == 'REQUESTED'
```

Comment:
I am deleting the status if it is 'SUCCESS' or 'FAILED' since both indicate that the asynchronous job has finished.